rust/src/storage/mod.rs

```diff
@@ -4,869 +4,142 @@
-        Ok(())
-    }
-
-    /// Removes all stored nasl_vts
-    pub fn clean_advisories(&self) -> Result<(), StorageError> {
-        let mut advisories = self.advisories.write()?;
-        advisories.clear();
-        Ok(())
-    }
-}
-
-impl Remover for DefaultDispatcher {
-    fn remove_kb(
-        &self,
-        key: &ContextKey,
-        kb_key: Option<String>,
-    ) -> Result<Option<Vec<Kb>>, StorageError> {
-        let mut kbs = self.kbs.write().unwrap();
-        Ok(match kb_key {
-            None => kbs
-                .remove(key)
-                .map(|x| x.values().flat_map(|x| x.clone()).collect()),
-            Some(x) => {
-                if let Some(kbs) = kbs.get_mut(key) {
-                    kbs.remove(&x)
-                } else {
-                    None
-                }
-            }
-        })
-    }
-
-    fn remove_result(
-        &self,
-        key: &ContextKey,
-        result_id: Option<usize>,
-    ) -> Result<Option<Vec<models::Result>>, StorageError> {
-        let mut results = self.results.write().unwrap();
-        if let Some(idx) = result_id {
-            if let Some(results) = results.get_mut(key.as_ref()) {
-                if let Some(idx) = results.iter().position(|x| x.id == idx) {
-                    return Ok(Some(vec![results.remove(idx)]));
-                }
-            }
-            Ok(None)
-        } else {
-            Ok(results.remove(key.as_ref()))
-        }
-    }
 }
 
-impl Dispatcher for DefaultDispatcher {
-    fn dispatch(&self, key: &ContextKey, scope: Field) -> Result<(), StorageError> {
-        match scope {
-            Field::NVT(x) => self.cache_nvt_field(key.as_ref(), x)?,
-            Field::KB(x) => self.cache_kb(key.clone(), x)?,
-            Field::NotusAdvisory(x) => {
-                if let Some(x) = *x {
-                    self.cache_notus_advisory(x)?
-                }
-            }
-            Field::Result(x) => self.cache_result(key.as_ref(), *x)?,
-        }
-        Ok(())
-    }
-
-    fn dispatch_replace(&self, key: &ContextKey, scope: Field) -> Result<(), StorageError> {
-        match scope {
-            Field::NVT(x) => self.cache_nvt_field(key.as_ref(), x)?,
-            Field::KB(x) => self.replace_kb(key, x)?,
-            Field::NotusAdvisory(x) => {
-                if let Some(x) = *x {
-                    self.cache_notus_advisory(x)?
-                }
-            }
-            Field::Result(x) => self.cache_result(key.as_ref(), *x)?,
-        }
-        Ok(())
-    }
-
-    fn on_exit(&self, _: &ContextKey) -> Result<(), StorageError> {
-        Ok(())
-    }
-}
-
-/// Holds iterator in memory
-pub struct InMemoryDataWrapper<T> {
-    inner: Box<dyn Iterator<Item = T>>,
-}
-
-impl<T> InMemoryDataWrapper<T>
-where
-    T: 'static,
+pub trait OspStorage: Retriever<Oid, Item = Nvt> + Retriever<Feed, Item = Vec<Nvt>> {}
+
+pub trait SchedulerStorage: Retriever<Oid, Item = Nvt> + Retriever<FileName, Item = Nvt> {}
```

Comment:
Although storage is not an own crate anymore I believe it should still be treated as an own module with own separation of concern on a functional level.

Therefore I think this should be moved into scheduling as it is just relevant there and not be defined in storage#mod.